src/components/numeral-date/numeral-date.test.tsx

```diff
@@ -0,0 +1,1995 @@
+    await user.type(dayInput, "32");
+    await user.type(yearInput, "2201");
+    await user.tab();
+
+    expect(
+      screen.getByText(
+        "Day should be a number within a 1-31 range. " +
+          "Year should be a number within a 1800-2200 range."
+      )
+    ).toBeVisible();
+  });
+
+  it("should render the appropriate validation message when the `Month` and `Year` fields are blurred and have values greater than the valid limits", async () => {
+    const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+    render(
+      <CarbonProvider validationRedesignOptIn>
+        <NumeralDate
+          value={{ dd: "", mm: "", yyyy: "" }}
+          onChange={() => {}}
+          enableInternalWarning
+        />
+      </CarbonProvider>
+    );
+
+    const monthInput = screen.getByRole("textbox", { name: "Month" });
+    const yearInput = screen.getByRole("textbox", { name: "Year" });
+    await user.type(monthInput, "13");
+    await user.type(yearInput, "2201");
+    await user.tab();
+
+    expect(
+      screen.getByText(
+        "Month should be a number within a 1-12 range. " +
+          "Year should be a number within a 1800-2200 range."
+      )
+    ).toBeVisible();
+  });
+
+  it("should render the appropriate validation message when the `Day`, `Month`, and `Year` fields are blurred and have values greater than the valid limits", async () => {
+    const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+    render(
+      <CarbonProvider validationRedesignOptIn>
+        <NumeralDate
+          value={{ dd: "", mm: "", yyyy: "" }}
+          onChange={() => {}}
+          enableInternalWarning
+        />
+      </CarbonProvider>
+    );
+
+    const dayInput = screen.getByRole("textbox", { name: "Day" });
+    const monthInput = screen.getByRole("textbox", { name: "Month" });
+    const yearInput = screen.getByRole("textbox", { name: "Year" });
+    await user.type(dayInput, "32");
+    await user.type(monthInput, "13");
+    await user.type(yearInput, "2201");
+    await user.tab();
+
+    expect(
+      screen.getByText(
+        "Day should be a number within a 1-31 range. " +
+          "Month should be a number within a 1-12 range. " +
+          "Year should be a number within a 1800-2200 range."
+      )
+    ).toBeVisible();
+  });
+});
+
+test("should submit the form when enter key is pressed", async () => {
+  const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+  const onSubmit = jest.fn();
+  render(
+    <form onSubmit={onSubmit}>
+      <NumeralDate
+        value={{ dd: "11", mm: "11", yyyy: "2011" }}
+        onChange={() => {}}
+      />
+      <button type="submit">Submit</button>
+    </form>
+  );
+
+  const dayInput = screen.getByRole("textbox", { name: "Day" });
+  await user.type(dayInput, "{enter}");
+
+  expect(onSubmit).toHaveBeenCalled();
+});
+
+test("should update the input values when the delete key is pressed", async () => {
```

Comment:
**comment** (non-blocking): I'm a bit surprised at this test. I've not seen in any other input components that we're testing that pressing delete clears the input - this is standard browser behaviour anyway, so I don't see what value it adds. As always I might be missing something specific to this component which gives it value.

Reply:
I agree it's odd, but we have code that implements it so imo it's better to test it now and remove if we refactor 

https://github.com/Sage/carbon/blob/3b62762ff577617f971f7fd36d9f84bf5ca468e1/src/components/numeral-date/numeral-date.component.tsx#L321